wgpu-hal/src/lib.rs

```diff
@@ -1242,8 +1242,38 @@ pub trait CommandEncoder: WasmNotSendSync + fmt::Debug {
 
     // pass common
 
-    /// Sets the bind group at `index` to `group`, assuming the layout
-    /// of all the preceding groups to be taken from `layout`.
+    /// Sets the bind group at `index` to `group`.
+    ///
+    /// If this is not the first call to `set_bind_group` within the current
+    /// render or compute pass:
+    ///
+    /// - If `layout` contains `n` bind group layouts, then any previously set
+    ///   bind groups at indices `n` or higher are cleared.
+    ///
+    /// - If the first `m` bind group layouts of `layout` are equal to those of
+    ///   the previously passed layout, but no more, then any previously set
+    ///   bind groups at indices `m` or higher are cleared.
```

Comment:
~~question: Shouldn't this be `any previously set bind groups at indices higher than `m` are cleared`, since the condition implies that index `m` is also still equal?~~

Answer from reading more thoroughly: We're using 0-based indices, so no, `m` is the index above the count.